controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -559,6 +559,42 @@ func (r *BareMetalHostReconciler) actionMatchProfile(prov provisioner.Provisione
 	return actionComplete{}
 }
 
+func (r *BareMetalHostReconciler) actionPreparing(prov provisioner.Provisioner, info *reconcileInfo) actionResult {
+	info.log.Info("preparing")
+
+	// Save provisioning settings.
+	provisioningSettings := info.host.Status.Provisioning.DeepCopy()
+	dirty, err := saveHostProvisioningSettings(info.host)
+	if err != nil {
+		return actionError{errors.Wrap(err, "Could not save the host provisioning settings")}
+	}
+
+	// Do prepare(manual clean).
+	provResult, started, err := prov.Prepare(dirty)
+	if err != nil {
+		return actionError{errors.Wrap(err, "error preparing host")}
+	}
+
+	if provResult.ErrorMessage != "" {
+		info.log.Info("handling cleaning error in controller")
```

Comment:
If I call `clearHostProvisioningSettings()` at the end of `actionDeprovisioning()`, this seems to cause the host's state to be unconditionally changed from "Ready" to "Preparing".

Reply:
> If I call `clearHostProvisioningSettings()` at the end of `actionDeprovisioning()`, this seems to cause the host's state to be unconditionally changed from "Ready" to "Preparing".

Yeah, that's expected and I think fine. We'll always want to redo the RAID config (if any) between provisionings because there's no guarantee that it hasn't been messed up.